### PR TITLE
Db version table

### PR DESCRIPTION
**Pull request by blueSwordfish**

### CODE REVIEW

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import gov.nih.ncats.common.util.TimeUtil;
+import ix.ginas.models.serialization.GsrsDateDeserializer;
+import ix.ginas.models.serialization.GsrsDateSerializer;
+import lombok.Data;
+
+@Entity
+@Data
+@Table(name = "ix_core_DB_GSRS_version")
```
Comment (tylerperyea) [inline gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:22]: Can we make all lowercase? Shouldn't matter but we always did all lowercase before

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+@Indexable(indexed = false)
+public class DBGSRSVersion {
+	
+	@Id    
+    @GeneratedValue(strategy = GenerationType.AUTO, generator = "LONG_SEQ_ID")
```
Comment (tylerperyea) [inline gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:27]: Mixed spaces and tabs. Can we use just spaces here?

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	
+	@Column(nullable=false)
+	String entity;
+	
+	@Column(nullable=false)
+	String version;
+	
+	@JsonSerialize(using = GsrsDateSerializer.class)
+    @JsonDeserialize(using = GsrsDateDeserializer.class)
+    public Date modified = TimeUtil.getCurrentDate();
```
Comment (tylerperyea) [inline gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:38]: Does this get a column or not? I don't know if I like that it always initalizes to now?

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	
+	String hash;
```
Comment (tylerperyea) [inline gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:40]: Does this get a column?

**Review by tylerperyea — CHANGES_REQUESTED** [review]

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java (hunk 4 above)
Comment (blueSwordfish) [reply gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:40]: yes, it gets a column

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java (hunk 1 above)
Comment (blueSwordfish) [reply gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:22]: Yes, they should all be lowercase.

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java (hunk 3 above)
Comment (blueSwordfish) [reply gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:38]: It gets a column. It is not fixed. It is the time of the insertion operation.

File: gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java (hunk 4 above)
Comment (epuzanov) [reply gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:40]: Do we realy need an additional Table for DB version tracking if we want use Liquibas? The Liquibase has its own special table for this.  https://docs.liquibase.com/concepts/tracking-tables/databasechangelog-table.html

Comment (blueSwordfish) [reply gsrs-core-entities/src/main/java/ix/core/models/DBGSRSVersion.java:39]: Yes, Liquibase has its own table. I think for the 3.1 release, we decided not to use Liquibase. 
We will need remove this part if we decide to use Liquibase in a future release.

**Pull request merged**